adal/src/telemetry/java/com/microsoft/aad/adal/HttpEvent.java

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package com.microsoft.aad.adal;
+
+import android.util.Pair;
+
+import java.net.URL;
+import java.util.List;
+import java.util.Map;
+
+class HttpEvent extends DefaultEvent {
+    HttpEvent(final String eventName) {
+        getEventList().add(Pair.create(EventStrings.EVENT_NAME, eventName));
+    }
+
+    void setUserAgent(final String userAgent) {
+        setEvent(EventStrings.HTTP_USER_AGENT, userAgent);
+    }
+
+    void setMethod(final String method) {
+        setEvent(EventStrings.HTTP_METHOD, method);
+    }
+
+    void setQueryParameters(final String queryParameters) {
+        setEvent(EventStrings.HTTP_QUERY_PARAMETERS, queryParameters);
+    }
+
+    void setResponseCode(final Integer responseCode) {
+        setEvent(EventStrings.HTTP_RESPONSE_CODE, responseCode.toString());
+    }
+
+    void setApiVersion(final String apiVersion) {
+        setEvent(EventStrings.HTTP_API_VERSION, apiVersion);
+    }
+
+    void setHttpPath(final URL httpPath) {
+        final String authority = httpPath.getAuthority();
+        final Discovery discovery = new Discovery();
+        if (!discovery.getValidHosts().contains(authority)) {
+            return;
+        }
+
+        final String[] splitArray = httpPath.getPath().split("/");
+
+        final StringBuilder logPath = new StringBuilder();
+        logPath.append(httpPath.getProtocol());
+        logPath.append("://");
+        logPath.append(authority);
+        logPath.append("/");
+
+        // we do not want to send tenant information
+        // index 0 is blank
+        // index 1 is tenant
+        for (int i = 2; i < splitArray.length; i++) {
+            logPath.append(splitArray[i]);
+            logPath.append("/");
+        }
+        setEvent(EventStrings.HTTP_PATH, logPath.toString());
```

Comment:
I have 2 concerns in here:
1. PII scrabing - are you sure that we would have tenant info always as a second parameter? Can we do something more predictive?
2. Why aren't we just wrote a small reg-ex to remove a second segement of the URL. It would be much cheaper and simpler that algorithm above.